### PR TITLE
[EISW-89820] [releases/2023/0] Rename VPUX to NPU

### PR DESCRIPTION
**Pull request by StefaniaHergane**
### Details:
 - Initial PR https://github.com/openvinotoolkit/openvino/pull/18974
 - Similar PR for `master` branch: https://github.com/openvinotoolkit/openvino/pull/19004

### Tickets:
 - EISW-89820

### CODE REVIEW

**Comment by PatrikStepan** [comment]
@ilya-lavrenov , @wangleis  please review this PR.

File: src/bindings/python/src/compatibility/openvino/requirements-dev.txt [hunk 1]
```diff
@@ -1 +1 @@
-cython>=0.29.32
+cython>=0.29.32,<=0.29.36
```
Comment (ilya-lavrenov) [inline src/bindings/python/src/compatibility/openvino/requirements-dev.txt:1]: please, revert changes in this file.

File: src/bindings/python/wheel/setup.py [hunk 2]
```diff
@@ -381,6 +382,11 @@ def generate_package(self, src_dirs):
 class CopyExt(build_ext):
     """Copy extension files to the build directory."""
 
+    user_options = [
+        ("skip-rpath", None, "Skips RPATH for Python extensions."),
+    ]
+    boolean_options = ["skip_rpath"]
+    
     def run(self):
         if len(self.extensions) == 1:
```
Comment (ilya-lavrenov) [inline src/bindings/python/wheel/setup.py:391]: please, revert all changes in this file.

File: thirdparty/CMakeLists.txt [hunk 3]
```diff
@@ -423,15 +437,16 @@ if(ENABLE_OV_PADDLE_FRONTEND OR ENABLE_OV_ONNX_FRONTEND OR ENABLE_OV_TF_FRONTEND
...
+        endif()
         # try to find newer version first (major is changed)
         # see https://protobuf.dev/support/version-support/ and
         # https://github.com/protocolbuffers/protobuf/commit/d61f75ff6db36b4f9c0765f131f8edc2f86310fa
-        find_package(Protobuf 4.22.0 QUIET NAMES ${find_package_protobuf_names})
+        find_package(Protobuf 4.22.0 QUIET ${protobuf_config})
         if(NOT Protobuf_FOUND)
             # otherwise, fallback to existing default
-            find_package(Protobuf 3.20.3 REQUIRED NAMES ${find_package_protobuf_names})
+            find_package(Protobuf 3.20.3 REQUIRED ${protobuf_config})
```
Comment (ilya-lavrenov) [inline thirdparty/CMakeLists.txt:449]: please, revert all changes in this file except VPUX -> NPU

File: cmake/templates/OpenVINOConfig.cmake.in [hunk 4]
```diff
@@ -375,10 +375,11 @@ endmacro()
...
 macro(_ov_find_protobuf_frontend_dependency)
     set(_OV_ENABLE_SYSTEM_PROTOBUF "@ENABLE_SYSTEM_PROTOBUF@")
-    # TODO: remove check for target existence
-    if(_OV_ENABLE_SYSTEM_PROTOBUF AND NOT TARGET protobuf::libprotobuf)
-        _ov_find_dependency(Protobuf @Protobuf_VERSION@ EXACT)
+    set(_OV_PROTOBUF_PACKAGE_CONFIG "@protobuf_config@")
+    if(_OV_ENABLE_SYSTEM_PROTOBUF)
+        _ov_find_dependency(Protobuf @Protobuf_VERSION@ EXACT ${_OV_PROTOBUF_PACKAGE_CONFIG})
     endif()
+    unset(_OV_PROTOBUF_PACKAGE_CONFIG)
```
Comment (ilya-lavrenov) [inline cmake/templates/OpenVINOConfig.cmake.in:382]: let's revert all changes in this file

**Review by ilya-lavrenov — CHANGES_REQUESTED** [review]

File: src/plugins/auto/src/plugin.cpp [hunk 5]
```diff
@@ -566,7 +566,7 @@ std::list<DeviceInformation> MultiDeviceInferencePlugin::GetValidDevice(
         }
     }
 
-    // Priority of selecting device: dGPU > VPUX > iGPU > MYRIAD > CPU
+    // Priority of selecting device: dGPU > NPU > iGPU > MYRIAD > CPU
     std::list<DeviceInformation> devices;
     if (networkPrecision == "INT8") {
         devices.splice(devices.end(), VPUX);
```
Comment (wangleis) [inline src/plugins/auto/src/plugin.cpp:572]: Please update device name from `VPUX` to `NPU`.

File: src/plugins/auto/src/plugin.cpp [hunk 6]
```diff
@@ -544,7 +544,7 @@ std::list<DeviceInformation> MultiDeviceInferencePlugin::GetValidDevice(
             MYRIAD.push_back(item);
             continue;
         }
-        if (item.deviceName.find("VPUX") == 0) {
+        if (item.deviceName.find("NPU") == 0) {
             VPUX.push_back(item);
```
Comment (wangleis) [inline src/plugins/auto/src/plugin.cpp:548]: Please update device name from `VPUX` to `NPU`.

File: src/plugins/auto/tests/unit/auto_runtime_fallback_test.cpp [hunk 7]
```diff
@@ -170,7 +170,7 @@ class AutoRuntimeFallback : public ::testing::TestWithParam<ConfigParams> {
                         std::this_thread::sleep_for(std::chrono::milliseconds(200));
                         return mockExeNetworkGPU_1; }));
         ON_CALL(*core, LoadNetwork(::testing::Matcher<const InferenceEngine::CNNNetwork&>(_),
-                    ::testing::Matcher<const std::string&>(StrEq(CommonTestUtils::DEVICE_KEEMBAY)),
+                    ::testing::Matcher<const std::string&>(StrEq(CommonTestUtils::DEVICE_NPU)),
                     ::testing::Matcher<const Config&>(_))).WillByDefault(InvokeWithoutArgs([this]() {
                         std::this_thread::sleep_for(std::chrono::milliseconds(200));
                         return mockExeNetworkVPUX; }));
```
Comment (wangleis) [inline src/plugins/auto/tests/unit/auto_runtime_fallback_test.cpp:176]: Please update device name from VPUX to NPU.

File: src/plugins/auto/tests/unit/auto_runtime_fallback_test.cpp [hunk 8]
```diff
@@ -275,7 +275,7 @@ TEST_P(AutoRuntimeFallback, releaseResource) {
                             std::this_thread::sleep_for(std::chrono::milliseconds(0));
                             return mockInferrequestGPU_1; }));
             }
-        } else if (deviceName == "VPUX") {
+        } else if (deviceName == "NPU") {
             mockInferrequestVPUX = std::make_shared<mockAsyncInferRequest>(
```
Comment (wangleis) [inline src/plugins/auto/tests/unit/auto_runtime_fallback_test.cpp:279]: Please update device name from VPUX to NPU.

File: src/plugins/auto/tests/unit/key_network_priority_test.cpp [hunk 9]
```diff
@@ -93,7 +93,7 @@ class KeyNetworkPriorityTest : public ::testing::TestWithParam<ConfigParams> {
                    StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _)).WillByDefault(RETURN_MOCK_VALUE(gpuCability));
        ON_CALL(*core, GetMetric(StrEq("MYRIAD"),
                    StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _)).WillByDefault(RETURN_MOCK_VALUE(myriadCability));
-       ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_KEEMBAY),
+       ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_NPU),
                    StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _)).WillByDefault(RETURN_MOCK_VALUE(vpuxCability));
```
Comment (wangleis) [inline src/plugins/auto/tests/unit/key_network_priority_test.cpp:97]: Please update device name from VPUX to NPU.

File: src/plugins/auto/tests/unit/parse_meta_device_test.cpp [hunk 10]
```diff
@@ -41,8 +41,8 @@ const char igpuFullDeviceName[] = "Intel(R) Gen9 HD Graphics (iGPU)";
 const char dgpuFullDeviceName[] = "Intel(R) Iris(R) Xe MAX Graphics (dGPU)";
 // const char myriadFullDeviceName[] = "Intel Movidius Myriad X VPU";
```
Comment (wangleis) [inline src/plugins/auto/tests/unit/parse_meta_device_test.cpp:42]: Please update device name in comments from VPUX to NPU

File: src/plugins/auto/tests/unit/select_device_test.cpp [hunk 11]
```diff
@@ -249,7 +249,7 @@ class SelectDeviceTest : public ::testing::TestWithParam<ConfigParams> {
                    StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _)).WillByDefault(RETURN_MOCK_VALUE(gpuCability));
        ON_CALL(*core, GetMetric(StrEq("MYRIAD"),
                    StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _)).WillByDefault(RETURN_MOCK_VALUE(myriadCability));
-       ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_KEEMBAY),
+       ON_CALL(*core, GetMetric(StrEq(CommonTestUtils::DEVICE_NPU),
                    StrEq(METRIC_KEY(OPTIMIZATION_CAPABILITIES)), _)).WillByDefault(RETURN_MOCK_VALUE(vpuxCability));
```
Comment (wangleis) [inline src/plugins/auto/tests/unit/select_device_test.cpp:253]: Please update device name in comments from VPUX to NPU.

File: tests/time_tests/test_runner/test_timetest.py [hunk 12]
```diff
@@ -43,7 +43,7 @@ def test_timetest(instance, executable, niter, cl_cache_dir, model_cache, model_
     :param niter: number of times to run executable
     :param cl_cache_dir: directory to store OpenCL cache
     :param cpu_cache: flag to enable model CPU cache
-    :param vpu_compiler: flag to change VPUX compiler type
+    :param vpu_compiler: flag to change NPU compiler type
```
Comment (wangleis) [inline tests/time_tests/test_runner/test_timetest.py:46]: Please update device name from VPUX to NPU.

File: tools/compile_tool/main.cpp [hunk 13]
```diff
@@ -353,7 +353,7 @@ bool isFP32(const ov::element::Type& type) {
 static void setDefaultIO(ov::preprocess::PrePostProcessor& preprocessor,
                          const std::vector<ov::Output<ov::Node>>& inputs,
                          const std::vector<ov::Output<ov::Node>>& outputs) {
-    const bool isVPUX = FLAGS_d.find("VPUX") != std::string::npos;
+    const bool isVPUX = FLAGS_d.find("NPU") != std::string::npos;
```
Comment (wangleis) [inline tools/compile_tool/main.cpp:356]: Please update device name from VPUX to NPU.

File: src/bindings/python/src/compatibility/openvino/requirements-dev.txt (hunk 1 above)
Comment (StefaniaHergane) [reply src/bindings/python/src/compatibility/openvino/requirements-dev.txt]: Done.

File: src/bindings/python/wheel/setup.py (hunk 2 above)
Comment (StefaniaHergane) [reply src/bindings/python/wheel/setup.py]: Done.

File: thirdparty/CMakeLists.txt (hunk 3 above)
Comment (StefaniaHergane) [reply thirdparty/CMakeLists.txt]: Should I revert the entire commit _Cherry-pick Support of protobuf >= 21_ https://github.com/openvinotoolkit/openvino/commit/f59ddf9a59c4c7877f40fdc5c14dacf9dec4d107 which updates 6 files? Or only the commit changes on this file `thirdparty/CMakeLists.txt`?

File: cmake/templates/OpenVINOConfig.cmake.in (hunk 4 above)
Comment (StefaniaHergane) [reply cmake/templates/OpenVINOConfig.cmake.in]: Done.

File: src/plugins/auto/src/plugin.cpp (hunk 5 above)
Comment (StefaniaHergane) [reply src/plugins/auto/src/plugin.cpp]: Done.

File: src/plugins/auto/src/plugin.cpp (hunk 6 above)
Comment (StefaniaHergane) [reply src/plugins/auto/src/plugin.cpp]: Done.

File: src/plugins/auto/tests/unit/auto_runtime_fallback_test.cpp (hunk 7 above)
Comment (StefaniaHergane) [reply src/plugins/auto/tests/unit/auto_runtime_fallback_test.cpp]: Done.

File: src/plugins/auto/tests/unit/parse_meta_device_test.cpp (hunk 10 above)
Comment (StefaniaHergane) [reply src/plugins/auto/tests/unit/parse_meta_device_test.cpp]: Done.

File: src/plugins/auto/tests/unit/select_device_test.cpp (hunk 11 above)
Comment (StefaniaHergane) [reply src/plugins/auto/tests/unit/select_device_test.cpp]: Done.

File: tests/time_tests/test_runner/test_timetest.py (hunk 12 above)
Comment (StefaniaHergane) [reply tests/time_tests/test_runner/test_timetest.py]: Done.

File: tools/compile_tool/main.cpp (hunk 13 above)
Comment (StefaniaHergane) [reply tools/compile_tool/main.cpp]: Done.

**Review by ilya-lavrenov — APPROVED** [review]

File: thirdparty/CMakeLists.txt (hunk 3 above)
Comment (StefaniaHergane) [reply thirdparty/CMakeLists.txt]: Clarified. Changes reverted after rebase.

**Review by wangleis — APPROVED** [review]

**Comment by ilya-lavrenov** [comment]
build_jenkins

**Comment by PatrikStepan** [comment]
We will revert DEVICE_KEEMBAY to DEVICE_NPU renaming for now to avoid cyclic dependency. This is not public UI, it is used only for internal tests, we can fix this later. This should unblock the PR checker.

**Comment by StefaniaHergane** [comment]
build_jenkins

**Pull request merged**